Solutions/Marain.ContentManagement.Api.Services/Marain/Cms/Api/Services/Dtos/CreateContentRequest.cs

```diff
@@ -0,0 +1,87 @@
+﻿// <copyright file="CreateContentRequest.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Cms.Api.Services.Dtos
+{
+    using System.Collections.Generic;
+    using System.Globalization;
+    using Corvus.Extensions.Json;
+
+    /// <summary>
+    /// A request to create a new content item in the CMS.
+    /// </summary>
+    public class CreateContentRequest
+    {
+        /// <summary>
+        /// Gets or sets the unique ID for the content.
+        /// </summary>
+        public string Id { get; set; }
+
+        /// <summary>
+        /// Gets or sets the category paths for the content.
+        /// </summary>
+        /// <remarks>
+        /// This supports multiple, hierarchical taxonomies for the content.
+        /// </remarks>
+        /// <seealso cref="Tags"/>
+        public IList<string> CategoryPaths { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tags for the content.
+        /// </summary>
+        /// <remarks>This support multiple, unstructured tags for the content.</remarks>
+        /// <seealso cref="CategoryPaths"/>
+        public IList<string> Tags { get; set; }
+
+        /// <summary>
+        /// Gets or sets the author of the content fragment.
+        /// </summary>
+        public CmsIdentity Author { get; set; }
+
+        /// <summary>
+        /// Gets or sets the title for the content.
+        /// </summary>
+        public string Title { get; set; }
+
+        /// <summary>
+        /// Gets or sets the description for the content.
+        /// </summary>
+        public string Description { get; set; }
+
+        /// <summary>
+        /// Gets or sets the culture for the content.
+        /// </summary>
+        public string Culture { get; set; }
```

Comment:
We have serialization support for CultureInfo (it serializes to/from a full named culture string) so you can use a CultureInfo here and avoid your special code in the `AsContent `method.

Reply:
But are we certain that the abstraction represented by `CultureInfo` is necessarily the same thing that this is meant to represent?

The docs for `CultureInfo.Equals` imply that it's possible to have two distinct `CultureInfo` objects representing the same named culture (same BCP-47 language string, e.g. `en-GB`) but which do not compare as equal:

> Two CultureInfo objects are considered equal if their Name and CompareInfo properties are equal.

I'm not actually sure under what circumstances you can get different a `CompareInfo` for two `CultureInfo` instances with the same name, but the wording of this strongly implies that it's possible. Is this mysterious difference captured by our serialization support? If not, how can we compare cultures? If the answer is "always use `CultureInfo.Name`" have we gained anything?

If your goal here is to identify a culture by its BCP-47 name, then I would argue that `string` is superior to `CultureInfo` because the problem with `CultureInfo` is that it is designed as a tool for inspecting or doing things with a culture; it is not designed to identify a culture (and the curiously qualified wording in the `Equals` docs imply that it might not work as you'd expect if you do try to use it for that).

If we were using Haskell or F#, we'd almost certainly use their abilities to say "a string, but specifically a string used for a particular purpose", so we could avoid the baggage of `CultureInfo` while having the benefit of a strong type that prevents you from accidentally using a string that represents something else. But we're not using those languages, so the question for me becomes: does the unambiguous, strongly-typed nature of `CultureInfo` outweigh the downside of the fact that it's not actually meant to act as an identifier?

Reply:
In this case, why can we not just pass the IEnumerable?

Reply:
The only place at the moment that we use the `CultureInfo` is in the Liquid rendering code, in `ContentDrop`, where it's used to back four properties that retrieve their values from the `CultureInfo`. That said, having poked through the rest of the code and the internals of the Liquid code, I can't see anywhere these are used, so it might be possible to get rid of them all - hopefully @mwadams can shed some light on whether these will be needed as the rendering work continues.

Either way, given it's only needed in the rendering code, I'm happy with keeping the `..Abstractions` code simple and using a string.

Reply:
On the dotnet side I can't think of a usecase where this is not going to be a CultureInfo.

Our serialization of that CultureInfo writes it to a string using `CultureInfo.Name`

To quote the docs, that is 

> The culture name in the format languagecode2-country/regioncode2, if the current CultureInfo is culture-dependent; or an empty string, if it's an invariant culture. languagecode2 is a lowercase two-letter code derived from ISO 639-1. country/regioncode2 is derived from ISO 3166 and usually consists of two uppercase letters, or a BCP-47 language tag.

It deserializes by parsing the string.

This leaves it in a decent string form for the JSON representation (and can be documented as such in the OpenAPI spec.)

So the question is really, but leaving it as a string, we are deferring its deserialization to the point of use. (Unless anyone can think of a usecase where we would *not* be using it as a CultureInfo).

My assumption is that most of the use cases would be in rendering scenarios where we want to understand what formatting to apply?

There is also the "finding the content for the locale" use case, but I am imagining we do that either a) by providing a URL-based approach (https://frob.com/fr/SomeContent) which I don't really like but is "normal" or, my preferred approach, creating a `ContentPayload` rather similar to the `ABTestSetPayload` which is something like a `LanguagePackPayload` which has `ContentReference`s to the content for different languages, and one or more fallback payloads which are surfaced at the one URL and selected based on the browser language or location or whatever. That would *not* require the CultureInfo to be a CI. But the benefits of not strongly typing would seem negligible.

Reply:
So, now I've thought about this again, it sounds like the use cases where it needs to be a `CultureInfo` are all rendering-related.

So the question becomes - is the main activity for this data reading and writing (i.e. the "content management" functions), or rendering.

If it's rendering then we should keep it as a `CultureInfo`, if it's management it should be a string.

On reflection I think we should be prioritising the rendering use case, so I'm going to leave it as a `CultureInfo`.